### PR TITLE
Add models for the Chi web framework

### PR DESCRIPTION
**Pull request by smowton**
This is mostly simple as the framework uses ordinary net/http methods and ordinary Go contexts for most purposes.

### CODE REVIEW

**Review by owen-mc — APPROVED** [review]
Once this is merged we should add Chi to the list of frameworks we support

File: ql/src/semmle/go/frameworks/Chi.qll [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+   * Functions that extract URL parameters, considered as a source of untrusted flow.
+   */
+  private class UserControlledFunction extends UntrustedFlowSource::Range, DataFlow::CallNode {
+    UserControlledFunction() {
+      this.getTarget().hasQualifiedName("github.com/go-chi/chi", ["URLParam", "URLParamFromCtx"])
+    }
+  }
+
+  /**
+   * Functions that extract URL parameters, considered as a source of untrusted flow.
```
Comment (max-schaefer) [inline ql/src/semmle/go/frameworks/Chi.qll:18]:
```suggestion
   * Methods that extract URL parameters, considered as a source of untrusted flow.
```

File: ql/test/library-tests/semmle/go/frameworks/Chi/vendor/github.com/go-chi/chi/stub.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
+// Code generated by depstubber. DO NOT EDIT.
+// This is a simple stub for github.com/go-chi/chi, strictly for use in testing.
+
+// See the LICENSE file for information about the licensing of the original library.
```
Comment (max-schaefer) [inline ql/test/library-tests/semmle/go/frameworks/Chi/vendor/github.com/go-chi/chi/stub.go:4]: Is the LICENSE file part of this PR?

**Review by max-schaefer — CHANGES_REQUESTED** [review]

**Comment by smowton** [comment]
@max-schaefer done

**Review by max-schaefer — APPROVED** [review]

**Pull request merged**